src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -43,6 +43,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for saga storage. Select another persistence or disable the sagas feature using endpointConfiguration.DisableFeature<Sagas>()");
             }
 
+            if (!context.Container.HasComponent<ISagaIdGenerator>())
```

Comment:
Putting my ServiceFabric TF hat on:

I'm sorry @andreasohlund but we just need saga id generation. Tackling persistence definition in general seems out of scope